refactor: centralize all download utils across app and apply special cloud-specific behavior

christian-byrne opened the pull request:
## Summary

Centralized all download functionalities across app. Then changed downloadFile on the cloud distribution to stream assets via blob fetches while desktop/local retains direct anchor downloads. This fixes issue where trying to download cross-origin resources opens them in the window, potentially losing the user's unsaved changes. 

## Changes

- **What**: Moved `downloadBlob` into `downloadUtil`, routed all callers (3D exporter, recording manager, node template export, workflow/palette export, Litegraph save, ~~`useDownload` consumers~~) through shared helpers, and changed `downloadFile` to `fetch` first when `isCloud` so cross-origin URLs download reliably
- `useDownload` is the exception since we simply cannot do model downloads through blob (forcing user to transfer the entire model data twice is bad). Fortunately on cloud, the user doesn't need to download models locally anyway.

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-6188-refactor-centralize-all-download-utils-across-app-and-apply-special-cloud-specific-behav-2946d73d365081de9f27f0994950511d) by [Unito](https://www.unito.io)

christian-byrne committed:
centralize download utils and use blob for cloud always

christian-byrne committed:
add knip ignore

jtydhr88 reviewed (dismissed):
LGTM

DrJKL commented on src/base/common/downloadUtil.ts:
```diff
@@ -15,15 +16,46 @@ export const downloadFile = (url: string, filename?: string): void => {
+  const link = document.createElement('a')
+  link.href = url
+  link.download = filename
+  link.style.display = 'none'
```
Why is this hidden but the one on 31 isn't?

DrJKL commented on src/base/common/downloadUtil.ts:
```diff
@@ -15,15 +16,46 @@ export const downloadFile = (url: string, filename?: string): void => {
+ * @param filename - The filename to suggest to the browser
+ * @param blob - The Blob to download
+ */
+export const downloadBlob = (filename: string, blob: Blob): void => {
```
Nit: Can we use `function`?
Function expressions in the codebase are one of the causes of HMR issues and at least one memory leak.
```suggestion
export function downloadBlob(filename: string, blob: Blob): void {
```

christian-byrne committed:
[refactor] extract link download logic and use function declarations - addresses review feedback

- Extract common triggerLinkDownload() helper to eliminate code duplication
- Convert arrow functions to function declarations to prevent HMR issues
- Make link hiding consistent by always using display: none

christian-byrne committed:
[test] optimize downloadUtil test structure - addresses review feedback

- Replace wildcard import with named import for better clarity
- Move URL spy stubs to module level for better performance
- Reset mocks in beforeEach instead of recreating them

christian-byrne commented on tests-ui/tests/base/common/downloadUtil.test.ts:
```diff
@@ -1,16 +1,40 @@
+import type { MockInstance } from 'vitest'
 
-import { downloadFile } from '@/base/common/downloadUtil'
+import * as downloadUtil from '@/base/common/downloadUtil'
```
Changed to named import in 0f77b46
